presto-memory/src/test/java/io/prestosql/plugin/memory/TestMemorySmoke.java

```diff
@@ -263,6 +264,40 @@ public void testSemiJoinDynamicFilteringBlockProbeSide()
                 ImmutableSet.of(1, ORDERS_COUNT, PART_COUNT));
     }
 
+    @Test
+    public void testCrossJoinDynamicFiltering()
+    {
+        assertUpdate("CREATE TABLE probe (k VARCHAR, v INTEGER)");
+        assertUpdate("CREATE TABLE build (vmin INTEGER, vmax INTEGER)");
+        assertUpdate("INSERT INTO probe VALUES ('a', 0), ('b', 1), ('c', 2), ('d', 3)", 4);
+        assertUpdate("INSERT INTO build VALUES (1, 2)", 1);
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v >= vmin", withBroadcastJoin(), 3, ImmutableSet.of(3, 1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v > vmin", withBroadcastJoin(), 2, ImmutableSet.of(2, 1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v <= vmax", withBroadcastJoin(), 3, ImmutableSet.of(3, 1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v < vmax", withBroadcastJoin(), 2, ImmutableSet.of(2, 1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v >= vmin AND v < vmax", withBroadcastJoin(), 1, ImmutableSet.of(1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v > vmin AND v <= vmax", withBroadcastJoin(), 1, ImmutableSet.of(1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v > vmin AND v < vmax", withBroadcastJoin(), 0, ImmutableSet.of(0, 1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v > vmin AND vmax < 0", withBroadcastJoin(), 0, ImmutableSet.of(0, 1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v >= vmin AND v <= vmax", withBroadcastJoin(), 2, ImmutableSet.of(2, 1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v BETWEEN vmin AND vmax", withBroadcastJoin(), 2, ImmutableSet.of(2, 1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v BETWEEN vmin AND vmax + 1", withBroadcastJoin(), 3, ImmutableSet.of(3, 1));
```

Comment:
From the test I don't know if `vmax + 1` filtered anything or not (that's why I suggested `vmin + 1`). Please change the test so that upper boundary also filters data.

Reply:
Good catch, thanks!
I've added more tests at `testCrossJoinDynamicFiltering` to reproduce #5777 and #5755 better.